lib/client/api.go

```diff
@@ -772,11 +773,18 @@ func RetryWithRelogin(ctx context.Context, tc *TeleportClient, fn func() error)
 	if err := tc.ActivateKey(ctx, key); err != nil {
 		return trace.Wrap(err)
 	}
+
+	// Attempt device login. This activates a fresh key if successful.
+	if err := tc.AttemptDeviceLogin(ctx, key); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
Shouldn't this be part of `TeleportClient.Login` rather than `RetryWithRelogin`, teleterm's `login` and tsh's `onLogin`? Do we ever want to skip this during `TeleportClient.Login`?

Reply:
That was my first attempt too, but I quickly ran into problems because tc.Login doesn't activate the keys, instead key activation is done by an explicit call to tc.ActivateKey - so that is the method we follow, instead of tc.Login.

The somewhat-nice side effect is that one has more programatic control over the API - say, for scenarios where one wants to authn the user but not the device, or, conversely, where one absolutely wants the device to be authenticated (via calling DeviceLogin). I can look into a refactor, but I'd rather push this one for later, if that's alright.